includes/multi-currency/AdminNotices.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * Admin notices for Multi-Currency.
+ *
+ * @package WooCommerce\Payments\MultiCurrency
+ */
+
+namespace WCPay\MultiCurrency;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class that will display admin notices.
+ */
+class AdminNotices {
+	/**
+	 * Notices.
+	 *
+	 * @var array
+	 */
+	private $notices = [];
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		add_action( 'admin_notices', [ $this, 'admin_notices' ] );
+		add_action( 'wp_loaded', [ $this, 'hide_notices' ] );
+	}
+
+	/**
+	 * Display any notices we've collected thus far.
+	 */
+	public function admin_notices() {
+		if ( ! current_user_can( 'manage_woocommerce' ) ) {
+			return;
+		}
+
+		$this->check_for_notices();
+
+		foreach ( (array) $this->notices as $notice_key => $notice ) {
+			echo '<div class="' . esc_attr( $notice['class'] ) . '" style="position:relative;">';
+
+			if ( $notice['dismissible'] ) {
+				?>
+				<a href="<?php echo esc_url( wp_nonce_url( add_query_arg( 'wcpay-multi-currency-hide-notice', $notice_key ), 'wcpay_multi_currency_hide_notices_nonce', '_wcpay_multi_currency_notice_nonce' ) ); ?>" class="woocommerce-message-close notice-dismiss" style="position:relative;float:right;padding:9px 0px 9px 9px 9px;text-decoration:none;"></a>
+				<?php
+			}
+
+			echo '<p>';
+			echo wp_kses(
+				$notice['message'],
+				[
+					'a' => [
+						'href'   => [],
+						'target' => [],
+					],
+				]
+			);
+			echo '</p></div>';
+		}
+	}
+
+	/**
+	 * Hides any admin notices.
+	 */
+	public function hide_notices() {
+		if ( isset( $_GET['wcpay-multi-currency-hide-notice'] ) && isset( $_GET['_wcpay_multi_currency_notice_nonce'] ) ) {
+			if ( ! wp_verify_nonce( wc_clean( wp_unslash( $_GET['_wcpay_multi_currency_notice_nonce'] ) ), 'wcpay_multi_currency_hide_notices_nonce' ) ) {
+				wp_die( esc_html__( 'Action failed. Please refresh the page and retry.', 'woocommerce-payments' ) );
+			}
+
+			if ( ! current_user_can( 'manage_woocommerce' ) ) {
+				wp_die( esc_html__( 'Cheatin&#8217; huh?', 'woocommerce-payments' ) );
+			}
+
+			$notice = wc_clean( wp_unslash( $_GET['wcpay-multi-currency-hide-notice'] ) );
+
+			switch ( $notice ) {
```

Comment:
It may be more appropriate to use an `if` statement here rather than a `switch`, since there's only one outcome we're checking for.

Reply:
Like the casting above, this was a `switch` in the original code to allow for matching several different notices and not just the one. I've updated it to an `if` now, and we can switch to a `switch` if/when needed. https://github.com/Automattic/woocommerce-payments/pull/2322/commits/b6ccb7d5165686c2bc4fdee0932a24f949d8fdfa

Reply:
Ah, that makes sense! Yeah, agreed that we can easily change back if needed if we have more options in future. :)